main/src/ca/uwaterloo/flix/language/phase/unification/zhegalkin/ZhegalkinExpr.scala

```diff
@@ -51,7 +52,7 @@ object ZhegalkinExpr {
       }
 
       // General case:
-      ZhegalkinExpr(cst, ts)
+      ZhegalkinExpr(cst, ts.sortBy(t => t.vars.size))
```

Comment:
Behavioral change :O 

Should this have a comment about why we sort?

Reply:
Its just for performance. Ill comment it in the next round.